lib/compiler/src/compile.erl

```diff
@@ -1563,6 +1567,31 @@ core_inline_module(Code0, #compile{options=Opts}=St) ->
 save_abstract_code(Code, St) ->
     {ok,Code,St#compile{abstract_code=erl_parse:anno_to_term(Code)}}.
 
+-define(META_DOC_CHUNK, <<"Docs">>).
+
+
+%% Adds documentation attributes to extra_chunks (beam file)
+beam_docs(Code, #compile{dir = Dir,
+                         extra_chunks = ExtraChunks}=St) ->
+    Docs = beam_doc:main(Dir, Code),
+    MetaDocs = [{?META_DOC_CHUNK, term_to_binary(Docs)} | ExtraChunks],
+    {ok, Code, St#compile{extra_chunks = MetaDocs}}.
+
+
+%% Strips documentation attributes from the code
+remove_doc_attributes(Code, St) ->
```

Comment:
If you do `module:module_info(attributes)`, the doc attributes are still listed.